### PR TITLE
etcdmain: infer bind addr from addr in v1 flagset

### PR DESCRIPTION
**Pull request by yichengq**

### CODE REVIEW

File: pkg/flags/flag.go [hunk 1]
```diff
@@ -85,6 +85,21 @@ func SetFlagsFromEnv(fs *flag.FlagSet) error {
 	return err
 }
 
+// InferBindAddrFromAddr infers the value of bindAddr flag from the value
+// of addr flag. Both flags' Value must be of type IPAddressPort. If the
+// bindAddr flag is set and the addr flag is unset, it will set the port of
+// bindAddr to the port of addr. Otherwise, it keeps the original values.
```
Comment (xiang90) [inline pkg/flags/flag.go]: it will set binAddr to 0:0:0:0:port

**Comment by yichengq** [comment]
refreshed.

**Comment by xiang90** [comment]
lgtm

**Comment by ArtfulCoder** [comment]
It seems that when I build using the build script, the generated etcd file is dynamically linked.
Can you verify that etcd is being statically linked.
There is an explicit comment in the build file that says that etcd is being statically linked. But that does not seem to be the case.

**Pull request merged**